stdlib/public/Darwin/Foundation/Data.swift

```diff
@@ -866,126 +806,141 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
             }
         }
 
-        @inlinable
-        var hashValue: Int {
-            let count: Int = numericCast(length)
-            return Swift.withUnsafeBytes(of: bytes) { (rawBuffer) -> Int in
-                return Int(bitPattern: CFHashBytes(UnsafeMutablePointer(mutating: rawBuffer.baseAddress?.assumingMemoryBound(to: UInt8.self)), count))
+        @inlinable @inline(__always) // This should always be inlined into _Representation.hash(into:).
```

Comment:
This (and the other internal hashing methods) should not need to be inlinable because they aren't public. Marking them inlinable isn't necessary for them to be inlined into the non-inlinable hasher code. Purely internal code can always be inlined.

Reply:
I can remove this if it's unnecessary, but without `@inline(__always)` we don't have a strict guarantee of this getting inlined... Because this can safely inline all the way up to `Data.hash(into:)`, is there any reason to not go for that guarantee?